### ISSUE TITLE
using circle ROIs and/or measurements in mrview

### ISSUE THREAD

**Issue by joelyss** [MRtrix3/mrtrix3#1961]
Hiya,

I was wondering if it is possible to take distance (mm) measurements or to use circular ROIs in mrview. 

For context, I have 2 tracts that I want to measure a set distance from each to locate the start of my ROI.

I have be doing all of my ROI drawing, reorientating and visualisation in mrview, so it would be great to keep it in this software if possible. Any ideas welcomed.

Many thanks,
Joely

**Comment by Lestropie** [comment]
Hi Joely,

> ... possible to take distance (mm) measurements ...

This at surface level sounds like #395. Would be a neat little thing to have, but it has not managed to make its way up anybody's priority list.

> .... or to use circular ROIs in mrview.

While it's possible to *use* spherical ROIs in `tckgen` and `tckedit`, they currently cannot be *visualised* in `mrview`. I have fairly recently made some progress on such - see #1836 - but it's not yet feature-complete.

> For context, I have 2 tracts that I want to measure a set distance from each to locate the start of my ROI.

There might be an alternative way of thinking about this problem.

*Somewhere* in my various collections of code snippets I have a stand-alone *MRtrix3* command called "`maskdistance`":

- It receives as input a mask image corresponding to some spatial marker, and a track file.

- For every input streamline, it calculates, for those voxels traversed by the streamline, the distance along the streamline trajectory between that voxel and the nearest location along the streamline that intersects the mask image.

- Finally, it generates a 3D image where the value in every voxel is the mean of these distances across the set of streamlines.

So the output is essentially, for every voxel, the *minimum WM pathway-based distance* between that voxel and the input mask image. This has the potential to be combined with the other data you have to derive "the start of your ROI" in something resembling a data-driven fashion, depending on the undisclosed details of your experiment.

If this is something that would actually be of utility outside of the project for which I wrote it, I could see about polishing it up and uploading as a Gist.

Rob

**Comment by joelyss** [comment]
Many thanks for the speedy response.

I had a look through the options and it seems to me that the best option would be a ruler tool. As my tracts do not pass through the ROI.

I also realised that I would come unstuck anyway as I would want to manually reorientate the slices and then draw the ROI which I think causes some issues in mrview as it wants to keep the ROI tool in the original orientation.

Really appreciate the help on this but I think I will have to perform this in another program and then come back to mrtrix3 for further tractography processing.

Out of interest, if I was to draw an ROI in say 3D slicer or fsleyes, which formats except nifti would mrview accept? I think 3D slicer's ROI save as .ascv. 

Many thanks,
Joely

**Comment by Lestropie** [comment]
> I also realised that I would come unstuck anyway as I would want to manually reorientate the slices and then draw the ROI which I think causes some issues in mrview as it wants to keep the ROI tool in the original orientation.

Yes, it automatically aligns specifically to the voxel grid of the ROI that is being edited. There's been plenty of requests to make the enhancements necessary to avoid that limitation (#1208).

> Out of interest, if I was to draw an ROI in say 3D slicer or fsleyes, which formats except nifti would mrview accept? I think 3D slicer's ROI save as .ascv.

The list of supported formats is [here](https://mrtrix.readthedocs.io/en/latest/getting_started/image_data.html#supported-image-formats). Was "`.acsv`" a typo? I can't find anything. Regardless, if 3D Slicer somehow can't export to 3D NIfTI, I would expect there to exist a conversion tool for whatever it produces; you won't be the first person wanting to get data from 3D Slicer into neuroimaging software :-P
